clang-tools-extra/clang-tidy/misc/AlphabeticalDeclOrderCheck.cpp

```diff
@@ -0,0 +1,84 @@
+//===--- AlphabeticalDeclOrderCheck.cpp - clang-tidy ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "AlphabeticalDeclOrderCheck.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+
+using namespace clang::ast_matchers;
+
+namespace clang {
+namespace tidy {
+namespace misc {
+
+void AlphabeticalDeclOrderCheck::registerMatchers(MatchFinder *Finder) {
+  Finder->addMatcher(declStmt().bind("stmt"), this);
+}
+
+bool AlphabeticalDeclOrderCheck::checkAlphabeticalOrder(const DeclStmt *MatchedStmt, std::vector<StringRef> &DeclNames, bool &fixDecl) {
```

Comment:
I don't have a strong opinion on this but perhaps consider `llvm::Expected<std::pair<std::vector<StringRef>, bool>>` or `llvm::Expected<AlphabeticalDeclOrderCheckResult>`. I think it's better if you can design in a way that avoids output parameters.